std/syscall.zig

```diff
@@ -268,6 +302,26 @@ pub fn write(fd: isize, buf: &const u8, count: isize) -> isize {
     syscall3(SYS_write, isize(fd), isize(buf), count)
 }
 
+pub fn open(path: []u8, flags: isize, perm: isize) -> isize {
+    syscall3(SYS_open, isize(&path[0]), flags, perm)
```

Comment:
path is `[]u8` which is not null terminated. Sadly I think we're going to have to do a memcpy for syscalls that require null terminated string parameters.